### PR TITLE
deprecate interactive API

### PR DESCRIPTION
**Pull request by vrancurel**
- Move the interactive_api examples under `openfl-tutorials/deprecated`
- Mark interactive_api as deprecated in the doc
- Remove interactive_api from CI/CD

### CODE REVIEW

File: openfl-tutorials/Federated_Pytorch_MNIST_Tutorial.ipynb
Comment (teoparvanov) [inline openfl-tutorials/Federated_Pytorch_MNIST_Tutorial.ipynb:1]: This file doesn't seem related to the Interactive API deprecation. You may have included it by mistake?

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -45,28 +45,8 @@ jobs:
...
-    - name: Interactive API - pytorch_kvasir_unet
-      run: |
-        python setup.py build_grpc
-        pip install torch==1.13.1
-        pip install torchvision==0.14.1
-        python -m tests.github.interactive_api_director.experiments.pytorch_kvasir_unet.run
-  
   cli:
-    needs: [lint, pytest-coverage, interactive-kvasir]
+    needs: [lint, pytest-coverage]
```
Comment (teoparvanov) [inline .github/workflows/ubuntu.yml:51]: I'd suggest including `workflow_interface_101_mnist` and `taskrunner` here - to highlight the shift towards Workflow API and TaskRunner API. 

WDYT?

**Review by teoparvanov — COMMENTED** [review]
LGTM at first glance. Here are a couple of things I noticed:

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -27,28 +27,8 @@ jobs:
...
-    - name: Interactive API - pytorch_kvasir_unet
-      run: |
-        python setup.py build_grpc
-        pip install torch==1.13.1
-        pip install torchvision==0.14.1
-        python -m tests.github.interactive_api_director.experiments.pytorch_kvasir_unet.run
-  
   cli: # from taskrunner.yml
-    needs: [pytest-coverage, interactive-kvasir]
+    needs: [pytest-coverage]
```
Comment (teoparvanov) [inline .github/workflows/windows.yml:35]: Same here, by the way - suggesting to run `workflow_interface_101_mnist` and `taskrunner` instead.

File: openfl-tutorials/Federated_Pytorch_MNIST_Tutorial.ipynb
Comment (vrancurel) [reply openfl-tutorials/Federated_Pytorch_MNIST_Tutorial.ipynb:1]: Correct, removed from commit

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (vrancurel) [reply .github/workflows/ubuntu.yml:51]: I will do another commit to reintroduce an interactive workflow example from new worflow API

File: .github/workflows/windows.yml (hunk 2 above)
Comment (vrancurel) [reply .github/workflows/windows.yml:35]: Same here

File: docs/about/releases.md [hunk 3]
```diff
@@ -11,7 +11,7 @@
...
 
-- **Expanded AI Accelerator Support**:  Intel® Data Center GPU Max Series support via the Intel® Extension for PyTorch, including examples for training on datasets such as [MNIST](https://github.com/securefederatedai/openfl/blob/develop/openfl-tutorials/experimental/104_MNIST_XPU.ipynb) (via Workflow API) and [TinyImageNet](https://github.com/securefederatedai/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_TinyImageNet_XPU) (via Interactive API) 
+- **Expanded AI Accelerator Support**:  Intel® Data Center GPU Max Series support via the Intel® Extension for PyTorch, including examples for training on datasets such as [MNIST](https://github.com/securefederatedai/openfl/blob/develop/openfl-tutorials/experimental/104_MNIST_XPU.ipynb) (via Workflow API) and [TinyImageNet](https://github.com/securefederatedai/openfl-contrib/tree/main/openfl_contrib_tutorials/interactive_api/PyTorch_TinyImageNet_XPU) (via Interactive API)
```
Comment (kminhta) [inline docs/about/releases.md]: From my comment on the [openfl-contrib PR](https://github.com/securefederatedai/openfl-contrib/pull/9#issuecomment-2463389082)

Would it be better to move these to a subfolder `openfl-tutorials/deprecated/` for v1.7 then remove outright in v1.6 instead of to the contrib repo?

**Comment by kminhta** [comment]
Thanks @vrancurel, this is definitely a much needed PR! However, I'd just like some clarity on the plan

**Review by MasterSkepticista — APPROVED** [review]

File: docs/about/releases.md (hunk 3 above)
Comment (teoparvanov) [reply docs/about/releases.md]: This is the suggestion indeed, based on our offline discussion.

**Review by tanwarsh — APPROVED** [review]

**Comment by vrancurel** [comment]
Thanks @tanwarsh please merge as needed.

**Comment by teoparvanov** [comment]
Hi @vrancurel , do you intend to refactor `kvasir-interactive` to Workflow API in this PR, or in a separate one?
If done separately, then this PR could be merged as-is. So, please let me know what your plan is.

**Review by kminhta — APPROVED** [review]

**Comment by vrancurel** [comment]
In a separate one, please merge @teoparvanov

**Review by teoparvanov — APPROVED** [review]
